apps/wallet-mobile/src/features/Send/common/useSendReceiver.tsx

```diff
@@ -0,0 +1,72 @@
+import {isDomain, isNameServer, isResolvableDomain, useResolverCryptoAddresses} from '@yoroi/resolver'
+import {Resolver} from '@yoroi/types'
+import * as React from 'react'
+import {useQueryClient} from 'react-query'
+
+import {debounceMaker} from '../../../utils/debounceMaker'
+import {useSend} from './SendContext'
+
+export const useSendReceiver = () => {
+  const queryClient = useQueryClient()
+
+  const {targets, selectedTargetIndex, receiverResolveChanged, addressRecordsFetched} = useSend()
+  const receiver = targets[selectedTargetIndex].receiver
+  const isUnsupportedDomain = !isResolvableDomain(receiver.resolve) && isDomain(receiver.resolve)
+
+  const {
+    error: receiverError,
+    cryptoAddresses,
+    refetch,
+    isLoading: isResolvingAddressess,
+    isSuccess,
+  } = useResolverCryptoAddresses(
+    {resolve: receiver.resolve},
+    {
+      enabled: false,
+    },
+  )
+  const debouncedRefetch = React.useMemo(() => debounceMaker(refetch, 300), [refetch])
+
+  const cancelPendingRequests = React.useCallback(
+    () => queryClient.cancelQueries({queryKey: ['useResolverCryptoAddresses']}),
+    [queryClient],
+  )
+
+  React.useEffect(() => {
+    if (receiver.as === 'domain') cancelPendingRequests().then(() => debouncedRefetch.call())
```

Comment:
if is domain abort sign abort pending requests, CNS module will have to expose it.